catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/BaseSolrCatalogProvider.java

```diff
@@ -494,7 +556,7 @@ private String getQuery(String attributeName, Set<String> ids) throws IngestExce
     return query;
   }
 
-  private String generatePrimaryKey() {
+  public String generatePrimaryKey() {
     return UUID.randomUUID().toString().replaceAll("-", "");
```

Comment:
what is the key used for?  this is out of scope, but we might want to hook this into the UuidGenerator service
https://github.com/codice/ddf/blob/master/platform/util/platform-util-uuidgenerator/util-uuidgenerator-api/src/main/java/org/codice/ddf/platform/util/uuidgenerator/UuidGenerator.java

Reply:
using to generate metacard id on ingest.